### PR TITLE
Add docbook-xsl-1.79.2

### PR DESCRIPTION
**Pull request by hattne**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hattne** [comment]
@conda-forge/staged-recipes: ready for review.

**Comment by hattne** [comment]
@conda-forge/staged-recipes: ping (ready for review).

**Review by hmaarrfk — CHANGES_REQUESTED** [review]
The installation and uninstallation seems to be getting done in the postlink and preunlink scripts. can this be avoided?

**Comment by hattne** [comment]
> Even given your explanation, There seems to be references to a webpage: `"http://cdn.docbook.org/release/xsl/"` in the post scripts. I do not think we should be referencing external websites during post scripts unless we are absolutely required to.

This is how the stylesheets are identified: the XSLT processor (_e.g._ `xsltproc`) is asked to apply a stylesheet (_e.g._ `http://cdn.docbook.org/release/xsl/current/html/docbook.xsl`) to some XML files. The processor then uses the catalog to map the URI to the locally installed stylesheets; there will be no need for the processor to actually visit the external site. Other stylsheets will be idintified by different URIs; they cannot be arbitrarily changed or removed.

**Comment by hattne** [comment]
> To me, it is unclear that the desired recipe will have the intended effect. I say this because the man page references an environment variable: `XML_CATALOG_FILES` which seems to be undefined on my conda installation. I have libxml

Yes, the compile-time default (`/etc/xml/catalog`) will not work with conda.  In my testing, I was working around that by setting `XML_CATALOG_FILES` while building; not pretty, I admit.

> is there a way to list the entries in the catalog? Perhaps through `xmlcatalog dump` ? we could add that as a test.

Excellent idea! I'll try to fix that.

> It may be that you should add activate/deactivate scripts to `libxml` to add and remove the XML_CATALOG_FILE respectively, and likely to create the default catalog file for you.

I had not thought of that!  Much nicer than setting it up each time it is needed. I'll submit a PR to `libxslt` tomorrow.

I think catalog-editing should stay in prelink and postunlink, because the catalog only changes when _this_ package (or similar packages providing other stylesheets) is installed/removed.

**Comment by hmaarrfk** [comment]
> I think catalog-editing should stay in prelink and postunlink, because the catalog only changes when this package (or similar packages providing other stylesheets) is installed/removed.

I understand. It might be that you can add "multiple catalogs" but i think that with a comment that repeats:
https://github.com/conda-forge/staged-recipes/pull/21087#issuecomment-1358745916

in the post/pre scripts it is more acceptable.

**Comment by hattne** [comment]
OK, done.

* Submitted [PR against `libxslt`](https://github.com/conda-forge/libxslt-feedstock/pull/36).  If and when that PR is merged, I'll revise this recipe to avoid setting the `XML_CATALOG_FILES` variable.
* Commented on the URI hardcoding
* Checking that the catalog has the relevant entry in the test.  This required moving the Windows tests to `run_tests.bat`; I moved the Unix tests for consistency.

**Comment by hmaarrfk** [comment]
Ocefpaf is quite active. It will be good to get his opinion on this. Let's give him 7 days?

**Comment by hattne** [comment]
> Ocefpaf is quite active. It will be good to get his opinion on this. Let's give him 7 days?

Certainly, no worries!

**Comment by hattne** [comment]
https://github.com/conda-forge/libxml2-feedstock/pull/91 was merged and it looks like it has not caused any trouble for a week. There is no need to set environment variables in the tests any longer—that is now handled by (de)activate scripts in `libxml2`.

@hmaarrfk: unless you have other concerns, I think this is finally good to go.

File: recipes/docbook-xsl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+requirements:
+  run:
+    # xmlcatalog is required by post-link.{bat,sh} and
+    # pre-unlink.{bat,sh}.
+    - libxml2
+
+test:
+  requires:
+    # Really need 2.10.3 build 6 or later.
+    - libxml2
```
Comment (hmaarrfk) [inline recipes/docbook-xsl/meta.yaml:26]: hmm this isn't really required here, but lets go with it.

Comment (hattne) [reply recipes/docbook-xsl/meta.yaml:26]: The idea really was to remember to depend on 2.10.4 or some such whenever that becomes available.

Comment (hmaarrfk) [reply recipes/docbook-xsl/meta.yaml:26]: you will want to make it a run constraint to be honest. Otherwise things like the migrator will have trouble.

Comment (hattne) [reply recipes/docbook-xsl/meta.yaml:26]: I see; I'll make a note fix this on the next update.

**Pull request merged**